nova/virt/vmwareapi/vmops.py

```diff
@@ -1537,77 +1537,164 @@ def _find_datastore_for_migration(self, instance, vm_ref, cluster_ref,
         if datastore_regex is not None:
             datastore_regex = re.compile(datastore_regex)
-        res_pool_ref = vm_util.get_res_pool_ref(self._session, cluster_ref)
+
+        res_pool_ref = server_data['res_pool']
+        res_pool_ref = vutil.get_moref(res_pool_ref, "ResourcePool")
         if res_pool_ref is None:
-            LOG.error("Cannot find resource pool", instance=instance)
+            LOG.error("Cannot find 1resource pool", instance=instance)
             raise exception.HostNotFound(hostg24=dest)
-        # find a datastore where the instance will be migrated to
-        ds = self._find_datastore_for_migration(instance, vm_ref, cluster_ref,
-                                                datastore_regex)
+
+        ds = server_data['datastore']
+        ds_ref = vutil.get_moref(ds, "Datastore")
+
         if ds is None:
             LOG.error("Cannot find datastore", instance=instance)
             raise exception.HostNotFound(host=dest)
-        LOG.debug("Migrating instance to datastore %s", ds.name,
-                  instance=instance)
-        # find ESX host in the destination cluster which is connected to the
-        # target datastore
-        esx_host = self._find_esx_host(cluster_ref, ds.ref)
+
+        esx_host = server_data['host']
+        host_ref = vutil.get_moref(esx_host, "HostSystem")
+
         if esx_host is None:
             LOG.error("Cannot find ESX host", instance=instance)
             raise exception.HostNotFound(host=dest)
 
         # Update networking backings
-
-        self._network_api = network.API()
-        network_info = self._network_api.get_instance_nw_info(context, instance)
-        client_factory = self._session.vim.client.factory
         devices = []
+
         hardware_devices = self._session._call_method(
             vutil, "get_object_property", vm_ref, "config.hardware.device")
-        image_meta = objects.ImageMeta.from_dict(
-            utils.get_image_from_system_metadata(
-                instance.system_metadata))
-        vif_model = image_meta.properties.get('hw_vif_model',
-                                              constants.DEFAULT_VIF_MODEL)
-        for vif in network_info:
-            vif_info = vmwarevif.get_vif_dict(
-                self._session, cluster_ref, vif_model, utils.is_neutron(), vif)
-            device = vmwarevif.get_network_device(hardware_devices,
-                                                  vif['address'])
-            devices.append(vm_util.update_vif_spec(client_factory, vif_info,
-                                                   device))
 
-        LOG.debug("Migrating instance to cluster '%s', datastore '%s' and "
-                  "ESX host '%s'", cluster_name, ds.name, esx_host,
-                  instance=instance)
+        for hardware_device in hardware_devices:
+            for device in hardware_device[1]:
+                if device.deviceInfo.label.find("Network adapter") > -1:
```

Comment:
Could you identify the network adapters differently? Perhaps over the type, or the backing.
The label could be changed, and in localised versions, they are different.